src/primitives/transaction_identifier.h

```diff
@@ -34,6 +34,7 @@ class transaction_identifier
 
 public:
     transaction_identifier() : m_wrapped{} {}
+    consteval explicit transaction_identifier(std::string_view hex_str) : m_wrapped{uint256{hex_str}} {}
```

Comment:
Would it be better to provide an `operator""_txid` and `_wtxid` along the lines of hexliterals in util/strencodings.h? They could be made available only to test code that way.

Adding `contstexpr` to the constructors in primitives/transaction_identifier.h and writing:

```c++
consteval auto operator""_txid(const char* str, size_t n) { return Txid::FromUint256(uint256{std::string_view{str, n}}); }
consteval auto operator""_wtxid(const char* str, size_t n) { return Wtxid::FromUint256(uint256{std::string_view{str, n}}); }

Wtxid wtxid_1{"85cd1a31eb38f74ed5742ec9cb546712ab5aaf747de28a9168b53e846cbda17f"_wtxid};
Txid txid_1{"bd0f71c1d5e50589063e134fad22053cdae5ab2320db5bf5e540198b0b5a4e69"_txid};
```

looks like it does the right thing, fwiw.

Reply:
Had a similar attempt in https://github.com/bitcoin/bitcoin/pull/31991#discussion_r1995691590 - what do you think @maflcko?